src/MyEventEmitter.js

```diff
@@ -1,12 +1,144 @@
+/* eslint-disable max-len */
+/* eslint-disable no-console */
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.listeners = {};
+  }
+
+  on(eventName, listener) {
+    if (!this.listeners[eventName]) {
+      this.listeners[eventName] = [];
+    }
+
+    if (!listener) {
+      const noListenerCase = () => {};
+
+      this.listeners[eventName].push(noListenerCase);
+
+      return;
+    }
+    this.listeners[eventName].push(listener);
+  }
+  once(eventName, listener) {
+    const resolveOnce = (...args) => {
+      if (listener) {
+        listener(...args);
+        this.off(eventName, listener);
+
+        return;
+      }
+      this.off(eventName, () => {});
+    };
+
+    this.on(eventName, resolveOnce);
+  }
+  off(eventName, listener) {
+    if (!listener) {
+      delete this.listeners[eventName];
+
+      return;
+    }
+
+    if (this.listeners[eventName]) {
+      const removalIndex = this.listeners[eventName]
+        .findIndex(curListener => curListener === listener);
+
+      this.listeners[eventName].splice(removalIndex, 1);
+    }
+  }
+  emit(eventName, ...args) {
+    if (this.listeners[eventName]) {
+      this.listeners[eventName].forEach(listener => listener(...args));
+    }
+  }
+  prependListener(eventName, listener) {
+    if (!this.listeners[eventName]) {
+      this.listeners[eventName] = [];
+    }
+
+    if (!listener) {
+      const noListenerCase = () => {};
+
+      this.listeners[eventName].unshift(noListenerCase);
+
+      return;
+    }
+    this.listeners[eventName].unshift(listener);
+  }
+  prependOnceListener(eventName, listener) {
+    const resolveOnce = (...args) => {
+      if (listener) {
+        listener(...args);
+        this.off(eventName, listener);
+
+        return;
+      }
+      this.off(eventName, () => {});
+    };
+
+    this.prependListener(eventName, resolveOnce);
+  }
+  removeAllListeners(eventName) {
+    if (!eventName) {
+      this.listeners = {};
+
+      return;
+    }
+
+    delete this.listeners[eventName];
+  }
+  listenerCount(eventName) {
+    if (this.listeners[eventName]) {
+      console.log(this.listeners[eventName].length);
+
+      return this.listeners[eventName].length;
+    }
+    console.log(0);
+
+    return 0;
+  }
 }
+
+const emitter = new MyEventEmitter();
+
+emitter.on('joke', (newJoke) => {
+  console.log(newJoke);
+});
+
+emitter.once('joke2', (newJoke) => {
+  console.log(newJoke);
+});
+
+emitter.prependListener('joke', (newJoke) => {
+  console.log(newJoke);
+});
```

Comment:
As per [docs](https://nodejs.org/api/events.html#emitteroneventname-listener), the calls should be chainable.
```suggestion
emitter.on('joke', (newJoke) => {
  console.log(newJoke);
})
  .once('joke2', (newJoke) => {
    console.log(newJoke);
  })
  .prependListener('joke', (newJoke) => {
    console.log(newJoke);
  });
```